extensions/amp-story/1.0/amp-story.js

```diff
@@ -1090,28 +1092,39 @@ export class AmpStory extends AMP.BaseElement {
   }
 
   /**
-   * Displays the unsupported browser UI: either the publisher provided UI, or
-   * fallbacks to a generic default.
+   * Displays changes to the support state of the browser
    * @param {boolean} isBrowserSupported
    * @private
    */
   onSupportedBrowserStateUpdate_(isBrowserSupported) {
-    if (isBrowserSupported) {
-      dev().error(TAG, 'No handler to exit unsupported browser state.');
-    }
-
     const fallbackEl = this.getFallback();
+    if (isBrowserSupported) {
+      // Removes the default unsupported browser layer or throws an error
+      // if the publisher has provided their own fallback
+      if (fallbackEl) {
+        dev().error(TAG, 'No handler to exit unsupported browser state on ' +
+        'publisher provided fallback.');
+      } else {
 
-    this.mutateElement(() => {
-      this.element.classList.add('i-amphtml-story-fallback');
-    });
-
-    // Displays the publisher provided fallback, or fallbacks to the default
-    // unsupported browser layer.
-    if (fallbackEl) {
-      this.toggleFallback(true);
+        removeElement(this.unsupportedBrowserLayer_.build());
+        this.mutateElement(() => {
+          this.element.classList.remove('i-amphtml-story-fallback');
+        });
+        this.continueAnyway = true;
+        this.layoutCallback();
```

Comment:
`layoutCallback()` (and other "callback" methods) generally shouldn't be called manually. Instead, can we refactor `layoutCallback`'s internals to avoid this? E.g. extract some logic into a private function and call that instead.

Reply:
Will do!